gdx-video-robovm/src/com/badlogic/gdx/video/VideoPlayerIos.java

```diff
@@ -0,0 +1,340 @@
+
+package com.badlogic.gdx.video;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.files.FileHandle;
+import com.badlogic.gdx.graphics.GL20;
+import com.badlogic.gdx.graphics.Pixmap;
+import com.badlogic.gdx.graphics.Texture;
+
+import org.robovm.apple.avfoundation.AVAsset;
+import org.robovm.apple.avfoundation.AVAssetTrack;
+import org.robovm.apple.avfoundation.AVMediaType;
+import org.robovm.apple.avfoundation.AVPlayer;
+import org.robovm.apple.avfoundation.AVPlayerItem;
+import org.robovm.apple.avfoundation.AVPlayerItemVideoOutput;
+import org.robovm.apple.avfoundation.AVPlayerStatus;
+import org.robovm.apple.coreaudio.AudioStreamBasicDescription;
+import org.robovm.apple.coremedia.CMAudioFormatDescription;
+import org.robovm.apple.coremedia.CMTime;
+import org.robovm.apple.coremedia.CMVideoCodecType;
+import org.robovm.apple.coremedia.CMVideoDimensions;
+import org.robovm.apple.coremedia.CMVideoFormatDescription;
+import org.robovm.apple.corevideo.CVImageBuffer;
+import org.robovm.apple.corevideo.CVPixelBuffer;
+import org.robovm.apple.corevideo.CVPixelBufferAttributes;
+import org.robovm.apple.corevideo.CVPixelBufferLockFlags;
+import org.robovm.apple.corevideo.CVPixelFormatType;
+import org.robovm.apple.foundation.NSArray;
+import org.robovm.apple.foundation.NSError;
+import org.robovm.apple.foundation.NSKeyValueChangeInfo;
+import org.robovm.apple.foundation.NSObject;
+import org.robovm.apple.foundation.NSURL;
+import org.robovm.objc.block.VoidBlock1;
+import org.robovm.objc.block.VoidBlock2;
+import org.robovm.objc.block.VoidBooleanBlock;
+import org.robovm.rt.bro.NativeObject;
+import org.robovm.rt.bro.ptr.BytePtr;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+
+/** iOS implementation of the VideoPlayer class.
+ *
+ * @author Maximilian Wende &lt;dasisdormax@mailbox.org&gt; */
+public class VideoPlayerIos implements VideoPlayer {
+
+	protected FileHandle file;
+	protected AVAsset asset;
+	AVPlayerItem playerItem;
+	AVPlayer player;
+	private boolean playerIsReady;
+	private boolean playerIsPrerolled;
+	private boolean pauseRequested;
+	private boolean isPlaying;
+	private boolean isLooping;
+
+	/* --- Video --- */
+	AVAssetTrack videoTrack;
+	CMVideoFormatDescription videoFormat;
+	CMVideoDimensions videoDimensions;
+	AVPlayerItemVideoOutput videoOutput;
+	private Texture texture;
+
+	/* --- Audio --- */
+	AVAssetTrack audioTrack;
+	CMAudioFormatDescription audioFormat;
+
+	/* --- Callbacks --- */
+	private VideoSizeListener videoSizeListener;
+	private CompletionListener completionListener;
+
+	protected AVPlayerItemVideoOutput createVideoOutput () {
+		CVPixelBufferAttributes attributes = new CVPixelBufferAttributes();
+		attributes.setPixelFormatType(CVPixelFormatType._24RGB);
+		return new AVPlayerItemVideoOutput(attributes);
+	}
+
+	protected void onVideoTrackLoaded (AVAssetTrack track) {
+		System.out.print("Track " + track.getTrackID() + ": ");
```

Comment:
I think we should log only via `Gdx.app.debug`

Reply:
Thanks, I forgot that previously.